### PR TITLE
Cleanup + add tests

### PR DESCRIPTION
**Pull request by bbelderbos**

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# dev
+black
+flake8
+httpx
+mypy
+pre-commit
+pytest
+pytest-cov
+pytest-env
+streamlit
```
Comment (bbelderbos) [inline requirements.in]: TODO: build a little front-end app to create code images ...

File: runtime.txt [hunk 2]
```diff
@@ -1 +1 @@
-python-3.9.9
+python-3.10.8
```
Comment (spackle0) [inline runtime.txt:1]: 👍

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+.PHONY: setup
+setup:
+	python3 -m venv venv && source venv/bin/activate && pip install -r requirements.txt
+
+.PHONY: lint
+lint:
+	flake8
```
Comment (spackle0) [inline Makefile:7]: Maybe add pylint here also explicitly.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+typing:
+	mypy tips tests
+
+.PHONY: runserver
+runserver:
+	uvicorn tips.main:app --reload
+
+.PHONY: cov
+cov:
+	pytest --cov=tips --cov-report=term-missing
```
Comment (spackle0) [inline Makefile]: Might want to add a percentage coverage here with `--cov-fail-under=MIN`, e.g. if you want to hit 80% you'd put 80 there.

Comment (bbelderbos) [reply Makefile]: TIL, thanks, added that.

File: Makefile (hunk 3 above)
Comment (bbelderbos) [reply Makefile:7]: Not used much, tends to be very verbose, but will try again ... thanks.

File: Makefile (hunk 4 above)
Comment (spackle0) [reply Makefile]: black should take care of most of it but yes, it can be verbose. I have a curated list of things to ignore in pylint.

Comment (bbelderbos) [reply Makefile]: Definitely want to adopt it because it looks at code quality as well, can you please share the things to ignore? Thanks!

**Pull request merged**